frostsnapp/rust/src/api/bitcoin.rs

```diff
@@ -166,6 +167,45 @@ pub struct Transaction {
 }
 
 impl Transaction {
+    pub(crate) fn from_template(tx_temp: &TransactionTemplate) -> Self {
+        let raw_tx = tx_temp.to_rust_bitcoin_tx();
+        let txid = tx_temp.txid();
+        let is_mine = tx_temp
+            .iter_locally_owned_inputs()
+            .map(|(_, _, spk)| spk.spk())
+            .chain(
+                tx_temp
+                    .iter_locally_owned_outputs()
+                    .map(|(_, _, spk)| spk.spk()),
+            )
+            .collect::<HashSet<_>>();
+        let prevouts = tx_temp
+            .inputs()
+            .iter()
+            .map(|input| (input.outpoint(), input.txout()))
+            .collect::<HashMap<bitcoin::OutPoint, bitcoin::TxOut>>();
+        Self {
+            inner: raw_tx,
+            txid: txid.to_string(),
+            confirmation_time: None,
+            last_seen: None,
+            prevouts,
+            is_mine,
+        }
+    }
+
+    pub(crate) fn fill_signatures(&mut self, signatures: &[EncodedSignature]) {
+        for (txin, signature) in self.inner.input.iter_mut().zip(signatures) {
```

Comment:
I think we should be iterating locally owned inputs here, otherwise we could be overwriting external signatures (a bug we have hit before iirc).

Could be worth considering whether it is appropriate to ~assert on the lengths (return None)  like you do in `attach_signatures_to_psbt`. 

Noticing (again) that transaction inputs have an implicit ordering that's preserved throughout the signing process, maintained by the signing coordinator when collecting signatures and handled in order through all signature collection points. I wonder if it is worth making less fragile at some point in the future.